posthog/caching/calculate_results.py

```diff
@@ -145,6 +146,12 @@ def calculate_for_query_based_insight(
     if isinstance(response, BaseModel):
         response = response.model_dump()
 
+    update_cached_state(  # Updating the relevant InsightCachingState
+        insight.team_id,
+        response.get("cache_key"),
+        response.get("last_refresh"),
+        result=None,  # Not caching the result here, since in HogQL this is the query runner's responsibility
+    )
```

Comment:
The only behavioral change in this PR: `TestDashboard::test_refresh_cache` revealed we weren't updating `InsightCachingState` in here. This change brings us fully back on track with tracking which insights need a background refresh